spec/root_spec.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'index route', type: :system do
+  it 'shows page content' do
+    visit '/'
+    expect(page).to have_content("Dick Johnson Is Dead") 
+    expect(page).to have_content("Blood & Water") 
+    expect(page).to have_content("Ganglands") 
+    expect(page).to have_content("Jailbirds New Orleans") 
+    expect(page).to have_content("Midnight Mass")
+    expect(page).to have_content("Sankofa")
```

Comment:
These tests seem to be testing state instead of logic. It might not be a bad thing at this point in time, but the test would also pass if in your controller, you rendered all first 10 titles in your CSV instead of reading from the CSV itself. As a more robust test, you could potentially stub `CSV` here, and have it return an array of 11 items, and then test that only the first 10 are displayed.

Because you haven't built a file parser here, testing this is a little bit more difficult than the way we do it in the codebase, but this test will definitely need to change as you start doing more complex things with your show data.

Reply:
> These tests seem to be testing state instead of logic. It might not be a bad thing at this point in time, but the test would also pass if in your controller, you rendered all first 10 titles in your CSV instead of reading from the CSV itself. As a more robust test, you could potentially stub `CSV` here, and have it return an array of 11 items, and then test that only the first 10 are displayed.
> 
> Because you haven't built a file parser here, testing this is a little bit more difficult than the way we do it in the codebase, but this test will definitely need to change as you start doing more complex things with your show data.

So basically in my latest build I haven't stubbed the file but I tested the logic of reading from the CSV itself